### PR TITLE
ci: replace CircleCI with Jenkins

### PR DESCRIPTION
**Pull request by smusali**
ci!: replace CircleCI with Jenkins

BREAKING CHANGE: Environment variables have been removed/renamed

This removes CircleCI and goes with Jenkins and sets up for the
2.0.0 release

Ref: #4

### CODE REVIEW

File: Jenkinsfile
Comment (darinspivey) [inline Jenkinsfile]: The current branch is still `master`.  We need to fix that later, though.

File: package.json
Comment (darinspivey) [inline package.json:32]: Remove `.taprc` if this block is in here.

File: package.json
Comment (darinspivey) [inline package.json]: This shouldn't be using `husky`.  It didn't really work for me anyway.

File: Jenkinsfile
Comment (smusali) [reply Jenkinsfile]: Ooops! FIxed!

File: package.json
Comment (smusali) [reply package.json:32]: Removed!

File: package.json
Comment (smusali) [reply package.json]: Removed!

File: Jenkinsfile
Comment (mdeltito) [inline Jenkinsfile]: `15` can be `16` now

File: package.json
Comment (mdeltito) [inline package.json]: missing a comma

File: tools/package.sh
Comment (mdeltito) [inline tools/package.sh:4]: I'd probably suggest moving the `npm ci --production` in here, rather than run it from the Jenkinsfile

File: package.json
Comment (mdeltito) [inline package.json]: we probably need to exclude this directory in `.gitignore` so it doesn't get added/committed as part of the release commit

File: package.json
Comment (mdeltito) [inline package.json]: nit: should be able to bump this to 5.x, and tap to 15.x as part of this, and `npm run lint:fix`. not necessary for this PR but would be a quick change in an additional commit if you wanted to include it.

**Review by mdeltito — CHANGES_REQUESTED** [review]
few comments, feel free to hold off until #1 lands and rebasing

File: package.json
Comment (smusali) [reply package.json]: @mdeltito, I added `**/*.zip` to `.gitignore` which handles this too

File: package.json
Comment (smusali) [reply package.json]: Updated their versions

File: tools/package.sh
Comment (smusali) [reply tools/package.sh:4]: Done! Added!

File: package.json
Comment (smusali) [reply package.json]: Fixed!

File: Jenkinsfile
Comment (smusali) [reply Jenkinsfile]: Fixed!

**Comment by darinspivey** [comment]
Build failing `tools/package.sh: line 4: zip: command not found`.  Does it *have* to be a zip file?  You've got `tar` in that image, but not sure if `.tgz` will work.

**Comment by mdeltito** [comment]
> Build failing `tools/package.sh: line 4: zip: command not found`. Does it _have_ to be a zip file? You've got `tar` in that image, but not sure if `.tgz` will work.

It does need to be a zip, there's a PR for adding `zip` to the ci images now

**Comment by smusali** [comment]
@darinspivey @mdeltito all looks good now!

**Review by darinspivey — DISMISSED** [review]

**Review by darinspivey — APPROVED** [review]

**Review by mdeltito — APPROVED** [review]

**Pull request merged**
